### PR TITLE
fix: Allow date as query, make locale configurable, relative time

### PR DESCRIPTION
**Pull request by DenverCoder1**
Features added:

* Optional query parameter for parsing a specific date using dateparser (e.g. `dt 2/25 3pm`)
* Support for choosing between `en-US` and `en-GB` locales
* Show relative time as relative (e.g. `in 5 minutes`) instead of the short time

Screenshot:

image

### CODE REVIEW

**Comment by DenverCoder1** [comment]
Sorry, I meant to open this is in my own fork, but if you'd like to merge my changes, feel free to let me know and I can reopen this.

**Comment by esfox** [comment]
I was actually starting to review the commits. lol
But if I may ask, is the default locale not dependent on the system's locale though?

**Comment by DenverCoder1** [comment]
Yes, it is.

I changed it to what I see in Discord, however maybe this could be user configurable in the extension?

**Comment by DenverCoder1** [comment]
I also made it use dateparser for allowing things like `dt 2/26/2020 3pm` to get a specific timestamp.

I'd be happy to contribute it to the original project if you are interested.

**Comment by esfox** [comment]
How would it be configurable then?
`dt` with an argument perhaps?

**Comment by DenverCoder1** [comment]
Maybe like a config.json file or something that the user of the extension could modify?

**Comment by DenverCoder1** [comment]
I've done something similar in this extension:
https://github.com/DenverCoder1/timezone-convert-albert-ext

**Comment by esfox** [comment]
Hmm this actually sounds useful.
If you can make a PR for it, that would be great but if not, I'll try to find time and see how it would be implemented using that extension you linked as a reference.

**Comment by DenverCoder1** [comment]
I can work on making the locale configurable and then add it to this PR

**Comment by DenverCoder1** [comment]
Also, I was planning to try making the relative timestamps show up with an accurate preview

**Comment by DenverCoder1** [comment]
I could also make the dateparser change in its own PR since I did make a separate branch for it

**Comment by esfox** [comment]
Yeah feel free to do any useful changes and we'll review these afterwards.

**Comment by DenverCoder1** [comment]
This branch already includes the code from #2, so to avoid merge conflicts, it might make more sense to just merge this one and close #2.

**Comment by DenverCoder1** [comment]
I have added a feature for displaying relative time strings to this PR as well since it will be easier than resolving conflicts later

File: __init__.py [hunk 1]
```diff
@@ -4,91 +4,177 @@
...
+from typing import List, Optional
+
+import albert
+import dateparser
 
 __title__ = "Discord Timestamp Generator"
-__version__ = "0.0.1"
+__version__ = "0.1.0"
 __triggers__ = ["dt"]
 __authors__ = "esfox"
```
Comment (esfox) [inline __init__.py]: @DenverCoder1 you can add yourself here as well. :)

**Comment by esfox** [comment]
I'm thinking maybe we could create another branch then merge it to that branch instead first so we can test it locally before merging it to main.

**Comment by DenverCoder1** [comment]
I've tested it locally using my fork. There should be a way for you to checkout my branch (I know in the GitHub Desktop application they allow switching branches to a Pull Request).

Though if it makes it easier for you, feel free to create another branch.

**Comment by esfox** [comment]
I made a new branch [here](https://github.com/esfox/albert-discord-timestamp/tree/dev).
Please make a new PR to that branch instead then we'll close this PR and merge the new one. :)

**Comment by DenverCoder1** [comment]
I've changed the base branch to dev, does that work for you?

**Comment by esfox** [comment]
Oh great that's better than. I was trying to change the target branch earlier but it just redirects me to your branch for some reason. Let's merge it. :)

Thanks a lot!

**Pull request merged**